pandas/core/arrays/datetimes.py

```diff
@@ -376,6 +377,16 @@ def _resolution(self):
     # ----------------------------------------------------------------
     # Array-like Methods
 
+    def __array__(self, dtype=None):
+        if is_object_dtype(dtype):
+            return np.array(list(self), dtype=object)
+        elif is_int64_dtype(dtype):
+            return self.asi8
```

Comment:
I think we can remove this elif branch. Numpy will afterwards convert the M8[ns] data to int, and in that way ensure the semantics of `np.asarray` regarding copy/no copy is followed.

Reply:
@jbrockmendel I opened https://github.com/pandas-dev/pandas/pull/23593, a PR doing the `__array__` for all datetimelike EAs, not only DatetimeArray (but so there is a bit of overlap with this PR)

Reply:
Great, I'll take a look at 23593.